lib/metric.js

```diff
@@ -1,16 +1,27 @@
 function Metric() {
   this.count = 0;
   this.time = 0;
+  this.running = 0;
   this.startTime = undefined;
 }
 
 Metric.prototype.start = function() {
   this.startTime = process.hrtime();
   this.count++;
+  this.running++;
 };
 
 Metric.prototype.stop = function() {
-  if (!this.startTime) return;
+  this.running--;
+
+  // Since we're only recording time once after all running locks are released,
+  // we're only collecting "wall clock" time and not "cost" time.
+  if (this.running > 0) {
+    return;
+  } else if (this.startTime === undefined) {
```

Comment:
Updated to handle this case. Also, removed the `startTime === undefined` check. That should never happen unless someone mucks with the value of `this.running`, in which case that's their own fault.